foojay-resolver/src/test/kotlin/org/gradle/toolchains/foojay/FoojayApiTest.kt

```diff
@@ -70,54 +76,74 @@ class FoojayApiTest {
+        assertMatchedDistributions(IBM_SEMERU, VENDOR_SPECIFIC, version, "Semeru")
 
-        assertEquals("Temurin", api.match(ADOPTIUM, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("Zulu", api.match(AZUL, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("Corretto", api.match(AMAZON, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("Liberica", api.match(BELLSOFT, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("Microsoft", api.match(MICROSOFT, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("Oracle OpenJDK", api.match(ORACLE, VENDOR_SPECIFIC, of(version))?.name)
-        assertEquals("SAP Machine", api.match(SAP, VENDOR_SPECIFIC, of(version))?.name)
+        assertMatchedDistributions(ADOPTIUM, VENDOR_SPECIFIC, version, "Temurin")
+        assertMatchedDistributions(AZUL, VENDOR_SPECIFIC, version, "Zulu")
+        assertMatchedDistributions(AMAZON, VENDOR_SPECIFIC, version, "Corretto")
+        assertMatchedDistributions(BELLSOFT, VENDOR_SPECIFIC, version, "Liberica")
+        assertMatchedDistributions(MICROSOFT, VENDOR_SPECIFIC, version, "Microsoft")
+        assertMatchedDistributions(ORACLE, VENDOR_SPECIFIC, version, "Oracle OpenJDK")
+        assertMatchedDistributions(SAP, VENDOR_SPECIFIC, version, "SAP Machine")
 
-        assertEquals("Graal VM CE $version", api.match(GRAAL_VM, VENDOR_SPECIFIC, of(version))?.name)
+        assertMatchedDistributions(GRAAL_VM, VENDOR_SPECIFIC, version, "Graal VM CE $version")
 
-        assertNull(api.match(APPLE, VENDOR_SPECIFIC, of(version))?.name)
-        assertNull(api.match(HEWLETT_PACKARD, VENDOR_SPECIFIC, of(version))?.name)
+        assertMatchedDistributions(APPLE, VENDOR_SPECIFIC, version)
+        assertMatchedDistributions(HEWLETT_PACKARD, VENDOR_SPECIFIC, version)
+    }
+
+    private fun assertMatchedDistributions(vendor: JvmVendorSpec, implementation: JvmImplementation, version: Int, vararg expectedDistributions: String) {
+        assertEquals(
+                listOf(*expectedDistributions),
+                api.match(vendor, implementation, of(version)).map { it.name },
+                "Mismatch in matching distributions for vendor: $vendor, implementation: $implementation, version: $version"
+        )
+    }
+
+    private fun assertMatchesDistributionsContainInOrder(vendor: JvmVendorSpec, implementation: JvmImplementation, version: Int, vararg expectedDistributions: String) {
+        val matches = api.match(vendor, implementation, of(version)).map { it.name }
+        assertTrue(
+                matches.containsAll(listOf(*expectedDistributions)),
+                "Mismatch in matching distributions for vendor: $vendor, implementation: $implementation, version: $version"
+        )
+        assertEquals(
+                matches.first(), expectedDistributions.first(),
+                "Wrong order in matching distributions for vendor: $vendor, implementation: $implementation, version: $version"
+        )
```

Comment:
This only checks the order for the first element but should check for the entire sub list.